pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -2118,6 +2129,28 @@ func (r *ReconcileClusterDeployment) updatePullSecretInfo(pullSecret string, cd
 	return true, nil
 }
 
+func (r *ReconcileClusterDeployment) ensurePrivateImagePullSecret(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (bool, error) {
+	secretName := os.Getenv(constants.HivePrivateImagePullSecret)
+	if secretName == "" {
+		cdLog.Debug("Hive private image pull secret missing or empty, ignoring...")
+		return false, nil
+	}
+
+	srcNamespace := controllerutils.GetHiveNamespace()
+	destNamespace := cd.GetNamespace()
+	cdLog.
+		WithField("secretName", secretName).
+		WithField("sourceNamespace", srcNamespace).
+		WithField("targetNamespace", destNamespace).
+		Debug("Ensuring hive private image pull secret exists")
+
+	src := types.NamespacedName{Name: secretName, Namespace: srcNamespace}
+	dest := types.NamespacedName{Name: secretName, Namespace: destNamespace}
+
+	// TODO: cross-NS ownership reference? is that even possible?
```

Comment:
It is not. However, we should parent it to the CD. There is precedent, and a handy helper function. See for example [how we do this](https://github.com/openshift/hive/blob/d09e43e7185bfc339838fc3ac0552adbcd4a83ab/pkg/controller/clusterdeployment/clusterdeployment_controller.go#L2353) for the pull secret.